### DISCUSSION TITLE
xls / xlsx / yaml is not supported

### DISCUSSION THREAD

**Discussion by mesa-git** [danielhrisca/asammdf#580] [Q&A] [answered]
Hey,

I recently reinstalled python and did the usual library instalations, and came up with this lines that were not here before.

```
xls is not supported
xlsx is not supported
yaml is not supported
```

that apparently come after the line 22 in v2_v3_blocks.py

`from .utils import get_fields, get_text_v3, MdfException, UINT16_u, UINT16_uf`

Is this something to be worried about? Am I missing a import? it seems to be working correctly, but not sure.

D:\repo_python\~Python3.9.7_old>python.exe -m pip list  
Package          Version  
asammdf          6.4.4  
attrs            21.2.0  
bitstruct        8.11.1
canmatrix        0.9.3
click            7.1.2
et-xmlfile       1.1.0
future           0.18.2
lxml             4.6.3
lz4              3.1.3
numexpr          2.7.3
numpy            1.21.2
opencv-python    4.5.3.56
openpyxl         3.0.9
pandas           1.3.3
pathlib2         2.3.6
Pillow           8.3.2
pip              21.2.4
PyQt5            5.15.4
pyqt5-plugins    5.15.4.2.2
PyQt5-Qt5        5.15.2
PyQt5-sip        12.9.0
pyqt5-tools      5.15.4.3.2
pyqtgraph        0.12.2
python-dateutil  2.8.2
python-docx      0.8.11
python-dotenv    0.19.0
pytz             2021.3
qt5-applications 5.15.2.2.2
qt5-tools        5.15.2.1.2
scipy            1.7.1
setuptools       58.2.0
six              1.16.0
wheel            0.37.0

**Comment by danielhrisca** [comment, marked as answer]
Nothing to worry about. Those messages come from the ``canmatrix`` package which has several dependencies for multiple file formats. (see https://github.com/ebroecker/canmatrix/blob/development/setup.py#L89)

**Reply by LaurentBerder** [reply]
Is there not a way to avoid this warning each time we load asammdf?
I personally got used to it, but moving programs in production with something like this does sound scary.

**Reply by danielhrisca** [reply]
You will need to install the dependencies pyaml, xlsxwriter, xlwt, openpyxl (I hope all are correct)

**Reply by danielhrisca** [reply]
check out the exact requirements specified in the ``canmatrix`` setup file https://github.com/ebroecker/canmatrix/blob/development/setup.py#L90

**Reply by LaurentBerder** [reply]
Thanks @danielhrisca

**Reply by 113haoyuansu** [reply]
ldf:ldfparser
xls:xlrd,lxwt
xlsx:xlsxwriter
yaml:pyyaml
Oh,my GOD!

**Reply by tonyqiuqiu** [reply]
A small correction: lxwt should be xlwt
